docs/addons/addon-migration-guide.mdx

```diff
@@ -17,79 +17,120 @@ Begin by updating your Storybook dependencies. Use the `next` tag for pre-releas
+3. Your addon's package size is minimized
 
-### React 18 for manager UI
+If your addon supports multiple major versions of Storybook, you can specify a wider version range in your peer dependencies:
 
-UI-based addons (e.g., [panels](../addons/addon-types.mdx#panels), [toolbars](../addons/addon-types.mdx#toolbars), [tabs](../addons/addon-types.mdx#tabs)) rely on React 18 to render their elements in the Storybook UI. Also, note that the `key` prop is no longer passed to the render function.
+```jsonc title="package.json"
+{
+  "name": "your-storybook-addon",
+  "peerDependencies": {
+    "storybook": "^8.0.0 || ^9.0.0"
+  },
+  "devDependencies": {
+    "storybook": "^9.0.0" // For local development
+  }
+}
+```
 
-### React peer dependency is no longer required
+However, we recommend releasing a new major version of your addon alongside new major versions of Storybook. This practice:
+1. Makes it easier to maintain your code
+2. Allows you to take advantage of new features and improvements
+3. Provides a clearer upgrade path for your users
 
-To remove your addon's peer dependency on React and reduce its install size, do the following:
+## Key changes for addons
 
-1. Move `react`, `react-dom`, and the globalized Storybook packages from `peerDependencies` to `devDependencies`
-2. Add the list of globalized packages to the `externals` property in the `tsup` configuration to ensure they are not part of the bundle.
+Here are the essential changes in version 9.0 that impact addon development.
 
-For an example, see [the updates we've made to the Addon Kit](https://github.com/storybookjs/addon-kit/compare/79282986..fa7ee7c). These changes are optional but recommended.
+### Package Consolidation
 
-<Callout variant="info">
-  This assumes your addon uses [tsup](https://tsup.egoist.dev/) for bundling. If your addon was built with an older version of the Addon Kit that uses Babel for bundling, you must first switch to tsup. For guidance, explore these older changes implemented in the Addon Kit [repository](https://github.com/storybookjs/addon-kit/pull/45/files).
-</Callout>
+Several packages have been consolidated into the main `storybook` package. Update your imports to use the new paths:
 
-### @storybook/components deprecations
+| Old Package              | New Path              |
+| ------------------------ | --------------------- |
+| @storybook/manager-api   | storybook/manager-api |
+| @storybook/preview-api   | storybook/preview-api |
+| @storybook/theming       | storybook/theming     |
+| @storybook/test          | storybook/test        |
+| @storybook/addon-actions | storybook/actions     |
 
-The `Icons` component from `@storybook/components` is now deprecated in favor of [`@storybook/icons`](https://github.com/storybookjs/icons). Additionally, various `Button` component props are also deprecated, with alternatives provided.
+Additionally, several internal packages have been moved under the `/internal` sub-path:
```

Comment:
We should explain the difference here. Something like "Internal paths are not public APIs so we don't guarantee they will stay stable according to semver. If you currently depend on them, we encourage finding replacements. However, if you just want the fastest way to upgrade, it's ok to keep using them. Just be forewarned!"